damnit/backend/db.py

```diff
@@ -210,10 +214,13 @@ def ensure_run(self, proposal: int, run: int, added_at: float=None, start_time:
     def change_run_comment(self, proposal: int, run: int, comment: str):
         self.set_variable(proposal, run, "comment", ReducedData(comment))
 
-    def add_user_variable(self, variable: UserEditableVariable, exist_ok=False):
+    def add_user_variable(self, variable: UserEditableVariable, exist_ok=False, overwrite=False):
         v = variable
+        if exist_ok and not overwrite and v.name in self.get_user_variables():
+            return
+
         with self.conn:
-            or_replace = ' OR REPLACE' if exist_ok else ''
+            or_replace = ' OR REPLACE' if overwrite else ''
```

Comment:
The `overwrite` parameter doesn't seem to be used anywhere?

If the variable specified already exists and `exist_ok` is True, I guess we should check that data type is the same between old & new variables, otherwise the DB can get in a mess. For stuff like the title & description, I'm inclined to think replacing the values with the new given ones is preferable to ignoring the new ones, but I don't feel strongly about that.

Reply:
Thanks. I think we should do this check also with `overwrite=False`, though, because it's also bad to silently ignore the new type and let the caller carry on as if it has added a new variable.

And do you foresee a need for the `overwrite` parameter? At present I think it's only used in tests, so if we don't have a use case for it, possibly we should get rid of it and have fewer code paths.

Reply:
Mmm I'm not sure I agree. If `overwrite=False` and `exist_ok=False` then the `INSERT` statement should throw an exception anyway, and if `overwrite=False` and `exist_ok=True` then I don't think an exception should be thrown if the type is different because to me `exist_ok=True` doesn't imply any kind of validation.

> And do you foresee a need for the `overwrite` parameter? At present I think it's only used in tests, so if we don't have a use case for it, possibly we should get rid of it and have fewer code paths.

Didn't you write it initially? :stuck_out_tongue: But yes, I can see us using it in the future when we allow modifying user-editable variables.

Reply:
> Didn't you write it initially?

I think you added `overwrite=` in this PR, no?

> to me exist_ok=True doesn't imply any kind of validation.

There's at least some precedent in that `Path('foo.txt').mkdir(exist_ok=True)` will throw an error if the path it's trying to create is something other than a directory. But OK, I don't feel strongly about this.